samples/lucy_utils.cpp

```diff
@@ -0,0 +1,365 @@
+            magfilt = TextureSampler::MagFilter::NEAREST;
+            matinstance = blur->createInstance();
+            matinstance->setParameter("weights", hweights, FILTER_SIZE);
+            break;
+        case VBLUR:
+            minfilt = TextureSampler::MinFilter::NEAREST;
+            magfilt = TextureSampler::MagFilter::NEAREST;
+            matinstance = blur->createInstance();
+            matinstance->setParameter("weights", vweights, FILTER_SIZE);
+            break;
+        case MIX:
+            minfilt = TextureSampler::MinFilter::LINEAR;
+            magfilt = TextureSampler::MagFilter::LINEAR;
+            matinstance = mix->createInstance();
+            matinstance->setParameter("secondary", secondary, TextureSampler(minfilt, magfilt));
+            break;
+    }
+    matinstance->setParameter("color", tex, TextureSampler(minfilt, magfilt));
+
+    Entity entity = EntityManager::get().create();
+    RenderableManager::Builder(1)
+        .boundingBox({{ 0, 0, 0 }, { 9000, 9000, 9000 }})
+        .material(0, matinstance)
+        .geometry(0, RenderableManager::PrimitiveType::TRIANGLES, vb, ib)
+        .build(*engine, entity);
+
+    return entity;
+}
+
+Entity createDisk(Engine* engine, Texture* reflection) {
+    constexpr int nslices = 64;
+    constexpr int nverts = nslices + 2;
+
+    static float4 verts[nverts];
+    int i = 0;
+    while (i < nslices + 1) {
+        float theta = 2.0f * M_PI * i / nslices;
+        verts[i++] = float4 {
+            std::cos(theta), std::sin(theta),
+            0.5 + 0.5 * std::cos(theta), 0.5 + 0.5 * std::sin(theta)
+        };
+    }
+    verts[i] = {0.0f, 0.0f, 0.5f, 0.5f};
+
+    static uint16_t indices[nslices * 3];
+    for (int i = 0, j = 0; i < nslices; ++i) {
+        indices[j++] = nverts - 1;
+        indices[j++] = i;
+        indices[j++] = (i + 1) % (nslices + 1);
+    }
+
+    VertexBuffer* vb = VertexBuffer::Builder()
+        .vertexCount(nverts)
+        .bufferCount(1)
+        .attribute(VertexAttribute::POSITION, 0, VertexBuffer::AttributeType::FLOAT2, 0, 16)
+        .attribute(VertexAttribute::UV0, 0, VertexBuffer::AttributeType::FLOAT2, 8, 16)
+        .build(*engine);
+
+    vb->setBufferAt(*engine, 0, VertexBuffer::BufferDescriptor(verts, sizeof(verts), nullptr));
+
+    IndexBuffer* ib = IndexBuffer::Builder()
+        .indexCount(nslices * 3)
+        .bufferType(IndexBuffer::IndexType::USHORT)
+        .build(*engine);
+
+    ib->setBuffer(*engine, IndexBuffer::BufferDescriptor(indices, sizeof(indices), nullptr));
+
+    filamat::Package pkg = filamat::MaterialBuilder()
+        .name("podium")
+        .material(R"SHADER(
+            void material(inout MaterialInputs material) {
+                vec3 n = texture(materialParams_normal, getUV0()).xyz * 2.0 - 1.0;
+
+                // The blue tiles normal map is very harsh, so we soften the normal vector.
+                material.normal = normalize(n + vec3(0, 0, 5));
```

Comment:
You should get used to adding `.0` to float constants, otherwise it own't compile for GLSL ES.

Reply:
This particular construction is legal in GLSL-ES: "If the basic type (bool, int, or float) of a parameter to a constructor does not match the basic type of the object being constructed, the scalar construction rules (above) are used to convert the parameters."